core/src/mock/mock_processor.cpp

```diff
@@ -96,11 +96,21 @@ namespace km {
                        std::u16string const & value)
     {
       auto i = _options.find(char16_t(scope) + key);
-      if (i == _options.end()) return option();
-
-      i->second = value;
-      persisted_store()[key] = value;
-      return option(scope, key, i->second);
+      if (i == _options.end()){
+        auto j = _options.emplace(char16_t(scope) + key,value);
+        if(j.second){
+          persisted_store()[key] = value;
+          return option(scope, key, j.first->second);
+        }
+        else {
+          return option();
+       }
+      }
```

Comment:
```suggestion
      if(i == _options.end()) {
        auto j = _options.emplace(char16_t(scope) + key, value);
        if(j.second) {
          persisted_store()[key] = value;
          return option(scope, key, j.first->second);
        }
        else {
          return option();
        }
      }
```

Can you also add a comment explaining what this is doing? The test for `j.second` is particularly obscure to me :grin: